lsat/interceptor.go

```diff
@@ -0,0 +1,184 @@
+package lsat
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"regexp"
+
+	"github.com/lightninglabs/loop/lndclient"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"github.com/lightningnetwork/lnd/zpay32"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	// CondPreimage is the caveat condition name for the payment preimage.
+	CondPreimage = "preimage"
+
+	// GRPCErrCode is the error code we receive from a gRPC call if the
+	// server expects a payment.
+	GRPCErrCode = codes.Internal
+
+	// GRPCErrMessage is the error message we receive from a gRPC call in
+	// conjunction with the GRPCErrCode to signal the client that a payment
+	// is required to access the service.
+	GRPCErrMessage = "payment required"
+
+	// AuthHeader is is the HTTP response header that contains the payment
+	// challenge.
+	AuthHeader = "WWW-Authenticate"
+
+	// MaxRoutingFee is the maximum routing fee in satoshis that we are
+	// going to pay to acquire an LSAT token.
+	// TODO(guggero): make this configurable
+	MaxRoutingFee = 10
```

Comment:
% or satoshis?

Reply:
The client library currently only supports fixed fee limit in sats. Added the suffix to make it clear.